intake/source/tests/test_json.py

```diff
@@ -0,0 +1,70 @@
+#-----------------------------------------------------------------------------
+# Copyright (c) 2012 - 2021, Anaconda, Inc. and Intake contributors
+# All rights reserved.
+#
+# The full license is in the LICENSE file, distributed with this software.
+#-----------------------------------------------------------------------------
+
+import json
+import os
+from pathlib import Path
+from typing import Dict, Optional
+
+import pytest
+from fsspec import open_files
+from intake.source.jsonfiles import JSONFileSource, JSONLinesFileSource
+
+here = os.path.abspath(os.path.dirname(__file__))
+
+EXTENSIONS = {
+    None: "",
+    "gzip": ".gz",
+    "bz2": ".bz2",
```

Comment:
@martindurant Is there a mapping somewhere from the compression string to the extension string? I'd love to use that, rather than this hand-written mapping, but couldn't find what I was looking for.

Reply:
`fsspec.utils.compressions` is the mapping {extension: internal name} and `fsspec.compression.compr` is the mapping {internal name: implementation function/class}. The latter will only have entries if the relevant library is installed.